cmd/server/cadence/server.go

```diff
@@ -220,7 +222,16 @@ func (s *server) startService() common.Daemon {
 		}
 	}
 
-	dispatcher, err := params.DispatcherProvider.Get(common.FrontendServiceName, s.cfg.PublicClient.HostPort)
+	// will return empty array if not enabled
+	var authProvider clientworker.AuthorizationProvider
+	if s.cfg.Authorization.OAuthAuthorizer.Enable {
+		privateKey, err := ioutil.ReadFile(s.cfg.Authorization.OAuthAuthorizer.JwtCredentials.PrivateKey)
+		if err != nil {
+			log.Fatalf("invalid private key path %s", s.cfg.Authorization.OAuthAuthorizer.JwtCredentials.PrivateKey)
+		}
+		authProvider = clientworker.NewJwtAuthorizationProvider(privateKey)
+	}
+	dispatcher, err := params.DispatcherProvider.Get(common.FrontendServiceName, s.cfg.PublicClient.HostPort, &client.DispatcherOptions{AuthProvider: &authProvider})
```

Comment:
This code is duplicated. Can we move it to a helper func in common/authorization pkg?